jenkins/scripts/integration_test.sh

```diff
@@ -72,27 +72,65 @@ then
 
   # Create test executer volume from copy of base volume
   echo "Creating a test executer volume from copy of base volume."
-  TEST_EXECUTER_VOLUME_ID="$(openstack volume create -f json \
-    --source "${BASE_VOLUME_NAME_ID}" \
-    "${TEST_EXECUTER_VM_NAME}" | jq -r '.id')"
+  create_test_executer_volume "${BASE_VOLUME_NAME_ID}" "${TEST_EXECUTER_VM_NAME}"
   
-  #--size 200 \
   # Wait for a test executer volume to be available...
   echo "Waiting for a test executer volume to be available."
+  retry=0
   until openstack volume show "${TEST_EXECUTER_VM_NAME}" -f json \
     | jq .status | grep "available"
   do
     sleep 10
+    # Check if test executer volume creation is failed
+    if [[ "$(openstack volume show "${TEST_EXECUTER_VM_NAME}" -f json \
+      | jq .status)" == "error"* ]];
+    then
+      echo "Test executer volume creation is failed"
+      # If test executer volume creation is failed, then retry volume creation only once
+      if [ $retry -eq 0 ]; then
+        echo "Deleting a test executer volume that failed to be created..."
+        openstack volume delete "${TEST_EXECUTER_VM_NAME}"
+        echo "Creating a fresh test executer volume..."
+        create_test_executer_volume "${BASE_VOLUME_NAME_ID}" "${TEST_EXECUTER_VM_NAME}"
+        retry=1
+      else
+        exit 1
+      fi
+      continue
+    fi
   done
-  
+
   echo "Resizing test executer volume."
   openstack volume set --size "${RESIZED_VM_SIZE}" "${TEST_EXECUTER_VM_NAME}"
   
   echo "Waiting for a resized test executer volume to be available."
+  retry=0
   until openstack volume show "${TEST_EXECUTER_VM_NAME}" -f json \
     | jq .size | grep "${RESIZED_VM_SIZE}"
   do
     sleep 10
+    if [[ "$(openstack volume show "${TEST_EXECUTER_VM_NAME}" -f json \
+      | jq .status)" == "error"* ]];
+    then
+      echo "Volume resizing is failed"
+      # If volume resizing is failed, then retry volume creation and resizing only once
+      if [ $retry -eq 0 ]; then
+        echo "Deleting a volume that failed to be resized..."
+        openstack volume delete "${TEST_EXECUTER_VM_NAME}"
+        echo "Creating a new test executer volume and resizing..."
+        create_test_executer_volume "${BASE_VOLUME_NAME_ID}" "${TEST_EXECUTER_VM_NAME}"
+        until openstack volume show "${TEST_EXECUTER_VM_NAME}" -f json \
+          | jq .status | grep "available"
+        do
+          sleep 10
+        done
```

Comment:
you probably need an escape condition here if the volume fails to create and ends in error state, to not only rely on the timeout.

Reply:
Done. I took completely each action of the test_integration script out for better readability. Please have a look @maelk !